File: SagradaFamilia/SagradaFamilia.Application/DTOs/Auth/MenuResponse.cs
namespace SagradaFamilia.Application.DTOs.Auth
{
    public class MenuResponse
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = string.Empty;
        public string? Icono { get; set; }
        public int Orden { get; set; }
        public List<OpcionResponse> Opciones { get; set; } = new();
    }
}

File: sagrada-familia-web/src/app/shared/interfaces/responses/menu.response.ts
export interface MenuResponse {
    id: number;
    nombre: string;
    icono: string;
    orden: number;
    opciones: OpcionResponse[];
}

export interface OpcionResponse {
    id: number;
    nombre: string;
    ruta: string;
    icono: string;
    orden: number;
    acciones: string[];
}